metricflow/model/objects/elements/identifier.py

```diff
@@ -41,31 +43,27 @@ class Identifier(HashableBaseModel):
     entity: Optional[str]
     identifiers: List[CompositeSubIdentifier] = []
     expr: Optional[str] = None
-
-    def __init__(  # type: ignore
-        self,
-        name: str,
-        type: IdentifierType,
-        role: Optional[str] = None,
-        entity: Optional[str] = None,
-        identifiers: Optional[List[CompositeSubIdentifier]] = None,
-        expr: Optional[str] = None,
-        **kwargs: Dict[str, Any],  # the parser may instantiate objects with additional fields (eg __parsing_context__)
-    ) -> None:
-        """Normal pydantic initializer except we set entity to name"""
-
-        identifiers = identifiers or []
-
-        super().__init__(
-            name=name,
-            type=type,
-            role=role,
-            entity=entity,
-            identifiers=identifiers,
-            expr=expr,
-        )
-        if self.entity is None:
-            self.entity = self.reference.element_name
+    metadata: Optional[Metadata]
+
+    @validator("entity", always=True)
+    @classmethod
+    def default_entity_value(cls, value: Any, values: Any) -> str:  # type: ignore[misc]
+        """Defaulting the value of the identifier 'entity' value using pydantic validator
+
+        If an entity value is provided that is a string, that will become the value of
+        entity. If the provifed entity value is None, the entity value becomes the
+        element_name representation of the identifier's name.
+        """
+
+        if value is None:
+            if "name" not in values:
+                raise ValueError("Failed to default entity value because objects name value was not defined")
+            return IdentifierReference(element_name=values["name"]).element_name
+
+        # guarantee value is string
+        if not isinstance(value, str):
+            raise ValueError(f"Entity value should be a string (str) type, but got {type(value)} with value: {value}")
```

Comment:
I think we can simplify to this:
```suggestion
        if value is None:
            if "name" not in values:
                raise ValueError("Failed to default entity value because objects name value was not defined")
            value = values["name"]

        # guarantee value is string
        if not isinstance(value, str):
            raise ValueError(f"Entity value should be a string (str) type, but got {type(value)} with value: {value}")
```

If "name" is not a string then Pydantic's validation should catch that, although I admit I don't know if there's an order in which these things get applied, so maybe an error message update would be useful.

Reply:
I agree that Pydantic's validations **should** catch when "name" isn't a string. But as you, I'm not sure if there is an order, and what it is, in which validations get applied, which is why I include the check. It should be unnecessary 🤷  but it also shouldn't slow things down dramatically.